zbus/tests/e2e.rs

```diff
@@ -699,6 +817,102 @@ async fn my_iface_test(conn: Connection, event: Event) -> zbus::Result<u32> {
     created_inside_proxy.ping().await?;
     proxy.destroy_obj("CreatedInside").await?;
 
+    // Test that interfaces emit signals when properties change
+    // according to their emits_changed_signal flags.
+    let mut props_changed = props_proxy.receive_properties_changed().await?;
+    let expected_property_value = 4;
+    proxy
+        .set_modifiable_default(expected_property_value)
+        .await?;
+    match props_changed.next().await {
+        Some(changed) => {
+            let expected_property_key = "ModifiableDefault";
+            let args = changed.args()?;
+            assert_eq!(
+                args.invalidated_properties(),
+                &vec![expected_property_key.to_string()]
+            );
+
+            let changed_prop_key = *args.changed_properties().keys().next().unwrap();
+            assert_eq!(changed_prop_key, expected_property_key);
+
+            let changed_value = &args.changed_properties()[expected_property_key];
+            assert_eq!(
+                <&Value as TryInto<u32>>::try_into(changed_value).unwrap(),
+                expected_property_value
+            );
+        }
+        None => panic!(""),
+    }
+    proxy.set_modifiable_true(expected_property_value).await?;
+    match props_changed.next().await {
+        Some(changed) => {
+            let expected_property_key = "ModifiableTrue";
+            let args = changed.args()?;
+            assert_eq!(
+                args.invalidated_properties(),
+                &vec![expected_property_key.to_string()]
+            );
+
+            let changed_prop_key = *args.changed_properties().keys().next().unwrap();
+            assert_eq!(changed_prop_key, expected_property_key);
+
+            let changed_value = &args.changed_properties()[expected_property_key];
+            assert_eq!(
+                <&Value as TryInto<u32>>::try_into(changed_value).unwrap(),
+                expected_property_value
+            );
+        }
+        None => panic!(""),
+    }
+    proxy
+        .set_modifiable_invalidates(expected_property_value)
+        .await?;
+    match props_changed.next().await {
+        Some(changed) => {
+            let expected_property_key = "ModifiableInvalidates";
+            let args = changed.args()?;
+            assert_eq!(
+                args.invalidated_properties(),
+                &vec![expected_property_key.to_string()]
+            );
+
+            assert_eq!(args.changed_properties().keys().len(), 0);
+        }
+        None => panic!(""),
```

Comment:
Might as well, just `unwrap` then and reduce code complexity? Same for other matches you're adding here.